src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs

```diff
@@ -157,38 +157,7 @@ internal sealed class ManagedDecryptorPal : DecryptorPal
                 }
                 else
                 {
-                    try
-                    {
-                        using (AsnWriter writer = new AsnWriter(AsnEncodingRules.BER))
-                        {
-                            writer.PushSequence();
-                            AsnReader reader = new AsnReader(decrypted, AsnEncodingRules.BER);
-
-                            while (reader.HasData)
-                            {
-                                writer.WriteEncodedValue(reader.GetEncodedValue());
-                            }
-
-                            writer.PopSequence();
-                            decrypted = writer.Encode();
-                        }
-                    }
-                    catch (CryptographicException)
-                    {
-                        // Decrypted data is not ASN.1 as expected.
-                        // For compatibility with Windows we will wrap it with the sequence tag
-                        // and create invalid sequence (content is not ASN.1)
-                        using (AsnWriter writer = new AsnWriter(AsnEncodingRules.BER))
-                        {
-                            // Writer does not have a way to create an invalid ASN.1.
-                            // We will encode it as octet string to bypass validation
-                            // and replace octet string tag (0x04) with sequence tag (0x30)
-                            writer.WriteOctetString(decrypted);
-                            decrypted = writer.Encode();
-                            Debug.Assert(decrypted[0] == 0x04);
-                            decrypted[0] = 0x30;
-                        }
-                    }
+                    decrypted = GetAsnSequenceWithContentNoValidation(decrypted);
```

Comment:
seems like we were always going into the catch path before so this got slightly smaller